### PR TITLE
ucr server environment

### PR DESCRIPTION
**Pull request by emord**
@snopoke icds-cas and icds-test exist in prod so it'll still try to load those configs. This adds an optional server environment to data source and report configs

@sravfeyn

### CODE REVIEW

File: corehq/apps/userreports/models.py [hunk 1]
```diff
@@ -525,6 +526,10 @@ def _all(cls):
     @classmethod
     def all(cls):
         for wrapped, path in cls._all():
+            if (wrapped.server_environment and
```
Comment (emord) [inline corehq/apps/userreports/models.py]: Eventually this should be required, but just want to get the icds reports off prod ES

File: corehq/apps/userreports/pillow.py [hunk 2]
```diff
@@ -115,9 +114,6 @@ def get_filtered_configs(self, configs=None):
         elif self.ucr_division:
             configs = _filter_by_hash(configs, self.ucr_division)
 
-        if self.filter_missing_domains:
-            configs = _exclude_missing_domains(configs)
```
Comment (snopoke) [inline corehq/apps/userreports/pillow.py]: can delete this function too

**Review by snopoke — APPROVED** [review]

File: corehq/apps/userreports/pillow.py (hunk 2 above)
Comment (emord) [inline corehq/apps/userreports/pillow.py]: https://github.com/dimagi/commcare-hq/pull/15547/commits/80946e83ebb8ca306c6a73993bbb4e400cceba4b

File: corehq/apps/userreports/models.py [hunk 3]
```diff
@@ -495,6 +495,7 @@ class StaticDataSourceConfiguration(JsonObject):
     """
     _datasource_id_prefix = STATIC_PREFIX
     domains = ListProperty()
+    server_environment = ListProperty()
```
Comment (sravfeyn) [inline corehq/apps/userreports/models.py]: (Minor), if we expect this to contain multiple environments, `server_environments` might be more suitable name.

**Review by sravfeyn — APPROVED** [review]

File: corehq/apps/userreports/models.py (hunk 3 above)
Comment (emord) [inline corehq/apps/userreports/models.py]: will make this change once the rest of hte ucrs get filled out with environments

**Pull request merged**